src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -74,11 +79,13 @@ public String getPartitionKey(String prefix, String path) {
   // Note that this does not test for duplicates. If invoked on an existing path it will overwrite
   // the entry. Existence checking is handled at upper layers.
   public void createDirectoryEntry(
-      TableServiceClient tableServiceClient, FireStoreDirectoryEntry createEntry) {
+      TableServiceClient tableServiceClient,
+      String collectionId,
```

Comment:
if `collectionId` is always a dataset ID or snapshot ID, why not make it a UUID here? I see that we're doing `datasetId.toString()` and `snapshotId.toString()` all over the place, so it might make sense to just take UUIDs all the way down until it actually needs to be turned into a String.

Reply:
That's a good point! I think I've changed them all - let me know if you see any I missed.